BootloaderCommonPkg/Include/Library/ContainerLib.h

```diff
@@ -0,0 +1,128 @@
+/** @file
+  Header file for container library implementation.
+
+  Copyright (c) 2019, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+#ifndef _CONTAINER_LIB_H_
+#define _CONTAINER_LIB_H_
+
+#include <Library/PcdLib.h>
+
+#define CONTAINER_LIST_SIGNATURE SIGNATURE_32('C','T','N', 'L')
+
+#define AUTH_TYPE_NONE                0
+#define AUTH_TYPE_SHA2_256            1
+#define AUTH_TYPE_SHA2_384            2
+#define AUTH_TYPE_SIG_RSA2048_SHA256  3
+#define AUTH_TYPE_SIG_RSA3072_SHA384  4
+
+typedef struct {
+  UINT32           Signature;
+  UINT32           HeaderCache;
+  UINT32           Base;
+  UINT32           Reserved;
+} CONTAINER_ENTRY;
+
+typedef struct {
+  UINT32           Signature;
+  UINT32           Reserved;
+  UINT32           TotalLength;
+  UINT32           Count;
+  CONTAINER_ENTRY  Entry[0];
+} CONTAINER_LIST;
+
+typedef struct {
+  UINT32           Signature;
+  UINT16           Version;
+  UINT16           DataOffset;
+  UINT32           DataSize;
+  UINT8            AuthType;
+  UINT8            ImageType;
+  UINT8            Flags;
+  UINT8            Count;
+} CONTAINER_HDR;
```

Comment:
container is very good for region support. lately if we added signed image using same header, do you think is it still a good name to be shared?

Reply:
CONTAINER is a generic term, I did not see the issue.